test/set.js

```diff
@@ -22,6 +22,22 @@ describe('set', function() {
     assert.deepEqual(b, {w: 1, x: {a: 5, b: 2, c: 3}, z: 4});
   });
 
+  it('handles empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({}, a);
+    assert.deepEqual(b, a);
+  });
+
+  it('handles nested empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({w: 0, x: {}, z: 3}, a);
+    assert.deepEqual(b, {w: 0, x: {a: 1, b: 2, c: 3}, z: 3});
+  });
+
```

Comment:
If we choose to interpret `{}` as a "flat" value that can override objects just like a primitive can then I don't see the issue. This is consistent with the rule "only recurse into things with keys". But the best thing would probably be to interpret empty objects as "deep" values and only primitives as flat values.

I don't see any of this as "significant API issues". The different behaviors are all explainable. They make sense. We just need to figure which is most expectable/useful.